File: schemaorg-rs/src/employer_aggregate_rating.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///https://schema.org/additionalType
///https://schema.org/Text
///https://schema.org/URL
pub type EmployerAggregateRatingAdditionalTypeFieldEnum = String;
///https://schema.org/EmployerAggregateRating
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct EmployerAggregateRating {
    #[serde(rename = "@context")]
    pub context: String,
    ///https://schema.org/sameAs
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///https://schema.org/name
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///https://schema.org/reviewAspect
    #[serde(rename = "reviewAspect")]
    #[serde_as(as = "OneOrMany<_>")]
    pub review_aspect: Vec<String>,
    ///https://schema.org/bestRating
    #[serde(rename = "bestRating")]
    #[serde_as(as = "OneOrMany<_>")]
    pub best_rating: Vec<EmployerAggregateRatingBestRatingFieldEnum>,
    ///https://schema.org/identifier
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<EmployerAggregateRatingIdentifierFieldEnum>,
    ///https://schema.org/author
    #[serde(rename = "author")]
    #[serde_as(as = "OneOrMany<_>")]
    pub author: Vec<EmployerAggregateRatingAuthorFieldEnum>,
    ///https://schema.org/potentialAction
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///https://schema.org/reviewCount
    #[serde(rename = "reviewCount")]
    #[serde_as(as = "OneOrMany<_>")]
    pub review_count: Vec<i32>,
    ///https://schema.org/ratingValue
    #[serde(rename = "ratingValue")]
    #[serde_as(as = "OneOrMany<_>")]
    pub rating_value: Vec<EmployerAggregateRatingRatingValueFieldEnum>,
    ///https://schema.org/worstRating
    #[serde(rename = "worstRating")]
    #[serde_as(as = "OneOrMany<_>")]
    pub worst_rating: Vec<EmployerAggregateRatingWorstRatingFieldEnum>,
    ///https://schema.org/mainEntityOfPage
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<EmployerAggregateRatingMainEntityOfPageFieldEnum>,
    ///https://schema.org/ratingCount
    #[serde(rename = "ratingCount")]
    #[serde_as(as = "OneOrMany<_>")]
    pub rating_count: Vec<i32>,
    ///https://schema.org/itemReviewed
    #[serde(rename = "itemReviewed")]
    #[serde_as(as = "OneOrMany<_>")]
    pub item_reviewed: Vec<Thing>,
    ///https://schema.org/disambiguatingDescription
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///https://schema.org/alternateName
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///https://schema.org/subjectOf
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<EmployerAggregateRatingSubjectOfFieldEnum>,
    ///https://schema.org/image
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<EmployerAggregateRatingImageFieldEnum>,
    ///https://schema.org/url
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///https://schema.org/additionalType
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<EmployerAggregateRatingAdditionalTypeFieldEnum>,
    ///https://schema.org/description
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<EmployerAggregateRatingDescriptionFieldEnum>,
    ///https://schema.org/ratingExplanation
    #[serde(rename = "ratingExplanation")]
    #[serde_as(as = "OneOrMany<_>")]
    pub rating_explanation: Vec<String>,
}

File: schemaorg-ts/src/employer_aggregate_rating.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type EmployerAggregateRatingAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/EmployerAggregateRating>
 **/
export interface EmployerAggregateRating {
  "@context": string;
  "@type": "EmployerAggregateRating";

  /**
   * <https://schema.org/reviewCount>
   **/
  reviewCount: all.OneOrMany<number>;

  /**
   * <https://schema.org/itemReviewed>
   **/
  itemReviewed: all.OneOrMany<all.Thing>;

  /**
   * <https://schema.org/ratingCount>
   **/
  ratingCount: all.OneOrMany<number>;

  /**
   * <https://schema.org/ratingValue>
   **/
  ratingValue: all.OneOrMany<all.EmployerAggregateRatingRatingValueFieldEnum>;

  /**
   * <https://schema.org/ratingExplanation>
   **/
  ratingExplanation: all.OneOrMany<string>;

  /**
   * <https://schema.org/worstRating>
   **/
  worstRating: all.OneOrMany<all.EmployerAggregateRatingWorstRatingFieldEnum>;

  /**
   * <https://schema.org/reviewAspect>
   **/
  reviewAspect: all.OneOrMany<string>;

  /**
   * <https://schema.org/bestRating>
   **/
  bestRating: all.OneOrMany<all.EmployerAggregateRatingBestRatingFieldEnum>;

  /**
   * <https://schema.org/author>
   **/
  author: all.OneOrMany<all.EmployerAggregateRatingAuthorFieldEnum>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.EmployerAggregateRatingMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.EmployerAggregateRatingIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.EmployerAggregateRatingDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.EmployerAggregateRatingSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.EmployerAggregateRatingAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.EmployerAggregateRatingImageFieldEnum>;
}
